### PR TITLE
feature/FOUR-12702: Start a Request: we need to enable only if the user was assigned "Start Permission"

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps

1. Create a process with start event
2. Assign a specific user or group 
3. Go to Processes and Start a request with a different user not assigned in the Start Permission
4. Go to Processes and Start a request with a different user not assigned in the group Start Permission
5. The START THIS PROCESS can not enable if the user was not assigned in the “Start Permission“

When the user does not have permission or the process does not have any start request to show use this style… disable the button

## Related Tickets & Packages
https://processmaker.atlassian.net/browse/FOUR-12702
But the PR solves the following tickets
- https://processmaker.atlassian.net/browse/FOUR-12913
- https://processmaker.atlassian.net/browse/FOUR-12788
- https://processmaker.atlassian.net/browse/FOUR-13089
- https://processmaker.atlassian.net/browse/FOUR-13142
- https://processmaker.atlassian.net/browse/FOUR-13144
- https://processmaker.atlassian.net/browse/FOUR-13140

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next
ci:APP_DEBUG=true

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-3a4af8a104.eng.processmaker.net

File: resources/js/processes-catalogue/components/ProcessOptions.vue [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <template>
   <div>
-    <buttons-start :process="process" />
+    <buttons-start :process="process"/>
```
Comment (fagubla) [inline resources/js/processes-catalogue/components/ProcessOptions.vue:3]: Expected a space before '/>'

**Review by fagubla — CHANGES_REQUESTED** [review]

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 2]
```diff
@@ -214,6 +216,83 @@ public function show(Request $request, Process $process)
...
+     *             @OA\Property(
+     *                 property="meta",
+     *                 type="object",
+     *                 ref="#/components/schemas/metadata",
+     *             ),
+     *         ),)
+     *     ),
+     * )
+     */
+    public function startEvents(Request $request, Process $process)
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:260]: Add the unit test related to this function

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 3]
```diff
@@ -1434,6 +1513,48 @@ protected function getRequestFilterBy(Request $request, array $searchableColumns
...
+    /**
+     * check if currentUser is member of a group
+     *
+     * @param object $event
+     * @param int $groupId
+     * @param Request $request
+     *
+     * @return bool
+     */
+    protected function checkUsersGroup($event, int $groupId, Request $request)
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:1525]: Add the unit test related to this function

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Review by fagubla — APPROVED** [review]

File: ProcessMaker/Http/Controllers/Api/ProcessController.php (hunk 2 above)
Comment (pmPaulis) [reply ProcessMaker/Http/Controllers/Api/ProcessController.php:260]: https://processmaker.atlassian.net/browse/FOUR-12702

File: ProcessMaker/Http/Controllers/Api/ProcessController.php (hunk 3 above)
Comment (pmPaulis) [reply ProcessMaker/Http/Controllers/Api/ProcessController.php]: https://processmaker.atlassian.net/browse/FOUR-12702

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-e25ca07418.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-e25ca07418.engk8s.processmaker.net

**Comment by pmPaulis** [comment]
I reviewed the following use cases:
https://github.com/ProcessMaker/processmaker/assets/42388723/5dc0d84b-3255-4a3a-8a6b-b71b50e9e7c6

**Review by pmPaulis — APPROVED** [review]

**Pull request merged**